File: schemas/crm/drizzle/tags.ts
// tags: reusable labels that can be applied to contacts, companies, and deals.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp } from "drizzle-orm/pg-core";

export const tags = pgTable(
  "tags",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull().unique(),
    color: text("color"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  }
);

File: schemas/email-campaign-marketing/spacetimedb/tags.rs
// tags: Labels that can be applied to contacts for categorization.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = tags, public)]
pub struct Tag {
    #[primary_key]
    pub id: String, // UUID

    #[unique]
    pub name: String,

    pub description: Option<String>,

    pub created_by: Option<String>, // UUID, FK → users.id (set null)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
